apps/liteprotocoltester/service_peer_management.nim

```diff
@@ -0,0 +1,218 @@
+
+    supportivePeers.delete(rndPeerIndex..rndPeerIndex)
+
+    let connOpt = pm.dialPeer(randomPeer, codec)
+    if (await connOpt.withTimeout(10.seconds)):
+      if connOpt.value().isSome():
+        found = some(randomPeer)
+        debug "Dialing successful",
+          peer = constructMultiaddrStr(randomPeer), codec = codec
+      else:
+        debug "Dialing failed", peer = constructMultiaddrStr(randomPeer), codec = codec
+    else:
+      debug "Timeout dialing service peer",
+        peer = constructMultiaddrStr(randomPeer), codec = codec
+
+  return found
+
+# Debugging PX gathered peers connectivity
+proc tryCallAllPxPeers*(
+    pm: PeerManager, codec: string, pubsubTopic: PubsubTopic
+): Future[Option[seq[RemotePeerInfo]]] {.async.} =
+  var capability = Capabilities.Filter
+  if codec.contains("lightpush"):
+    capability = Capabilities.Lightpush
+  elif codec.contains("filter"):
+    capability = Capabilities.Filter
+
+  var supportivePeers = pm.wakuPeerStore.getPeersByCapability(capability)
+
+  lpt_px_peers.set(supportivePeers.len)
+  debug "Found supportive peers count", count = supportivePeers.len()
+  debug "Found supportive peers", supportivePeers = $supportivePeers
+  if supportivePeers.len == 0:
+    return none(seq[RemotePeerInfo])
+
+  var okPeers: seq[RemotePeerInfo] = @[]
+
+  while supportivePeers.len > 0:
+    let rndPeerIndex = rand(0 .. supportivePeers.len - 1)
+    let randomPeer = supportivePeers[rndPeerIndex]
+
+    debug "Dialing random peer",
+      idx = $rndPeerIndex, peer = constructMultiaddrStr(randomPeer)
+
+    supportivePeers.delete(rndPeerIndex, rndPeerIndex)
+
+    let connOpt = pm.dialPeer(randomPeer, codec)
+    if (await connOpt.withTimeout(10.seconds)):
+      if connOpt.value().isSome():
+        okPeers.add(randomPeer)
+        info "Dialing successful",
+          peer = constructMultiaddrStr(randomPeer), codec = codec
+        lpt_dialed_peers.inc()
+      else:
+        lpt_dial_failures.inc()
+        error "Dialing failed", peer = constructMultiaddrStr(randomPeer), codec = codec
+    else:
+      lpt_dial_failures.inc()
+      error "Timeout dialing service peer",
+        peer = constructMultiaddrStr(randomPeer), codec = codec
+
+  var okPeersStr: string = ""
+  for idx, peer in okPeers:
+    okPeersStr.add(
+      "    " & $idx & ". | " & constructMultiaddrStr(peer) & " | protos: " &
+        $peer.protocols & " | caps: " & $peer.enr.map(getCapabilities) & "\n"
+    )
+  echo "PX returned peers found callable for " & codec & " / " & $capability & ":\n"
+  echo okPeersStr
+
+  return some(okPeers)
+
+proc pxLookupServiceNode*(
+    node: WakuNode, conf: LiteProtocolTesterConf
+): Future[Result[bool, void]] {.async.} =
+  var codec: string = WakuLightPushCodec
+  if conf.testFunc == TesterFunctionality.RECEIVER:
+    codec = WakuFilterSubscribeCodec
+
+  if node.wakuPeerExchange.isNil():
+    let peerExchangeNode = translateToRemotePeerInfo(conf.bootstrapNode).valueOr:
+      return err()
```

Comment:
In this case, I think we can bring better feedback. Applies elsewhere within this module.